lib/src/main/java/ca/jbrains/pos/PointOfSale.java

```diff
@@ -1,15 +1,37 @@
 package ca.jbrains.pos;
 
+import ca.jbrains.pos.domain.*;
+import ca.jbrains.pos.stub.StubProductFound;
+import io.vavr.control.Either;
+
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.util.stream.Stream;
 
 public class PointOfSale {
     public static void main(String[] args) {
-        parseInput(new InputStreamReader(System.in)).forEachOrdered(
-                line -> displayToConsole(displaySellOneItem(line, (ignored) -> "::a hardcoded response for every barcode::"))
-        );
+        parseInput(new InputStreamReader(System.in))
+                .map(PointOfSale::parseCommand)
+                .forEachOrdered(PosCommand::execute);
+    }
+
+    private static PosCommand parseCommand(String barcodeInput) {
```

Comment:
I notice duplication at the end of both paths: `displayToConsole()`. Try to remove it.

Reply:
right, done